jenkins/ci.suse.de/openstack-submit-project.yaml

```diff
@@ -25,90 +25,83 @@
 
     builders:
       - shell: |
-          # DEBUG MODE with E=echo
-          # NORMAL MODE with E=
+          # to debug set E=echo
           E=
 
-          if [ -z "$starttime" ] ; then
+          if [[ ! $starttime ]] ; then
               echo "FATAL: calling script needs to specify starttime"
               exit 1
           fi
 
           case $subproject in
-          None) echo "No valid 'subproject' provided - defaults for safety reasons to 'None'"
-                exit 1
-          ;;
-          "-")  echo "Subproject set to '-': skip job, do not fail (eg. for Master project)"
-                exit 0
-          ;;
-          esac
-
-          export BS=IBS
-          case $BS in
-          OBS)
-              OSC="osc -A https://api.opensuse.org"
-          ;;
-          IBS)
-              OSC="osc -A https://api.suse.de"
-          ;;
+              None) echo "No valid 'subproject' provided - defaults for safety reasons to 'None'"
+                    exit 1
+              ;;
+              "-")  echo "Subproject set to '-': skip job, do not fail (eg. for Master project)"
+                    exit 0
+              ;;
           esac
 
+          BS=IBS
+          OSC="osc -A https://api.suse.de"
 
           COS=$project
           ECOS=""
           COSS=$COS:$subproject
-          export BS_CHECKOUT=/home/jenkins/${BS}_CHECKOUT/$COSS
+          BS_CHECKOUT=/home/jenkins/${BS}_CHECKOUT/$COSS
           submitcmd="$OSC copypac -K -e"
           OSCBUILDDIST=SLE_12_SP1
 
           case $project in
-          Devel:Cloud:6)
-              ECOS=systemsmanagement:crowbar:3.0
-          ;;
-          Devel:Cloud:6:SAP)
-              ECOS=systemsmanagement:crowbar:3.1
-          ;;
-          Devel:Cloud:7)
-              OSCBUILDDIST=SLE_12_SP2
-              ECOS=systemsmanagement:crowbar:4.0
-          ;;
-          Devel:Cloud:8)
-              OSCBUILDDIST=SLE_12_SP3
-              ECOS=systemsmanagement:crowbar:5.0
-          ;;
-          *) echo "Please add support for the project: $project"
-              exit 1
+              Devel:Cloud:6)
+                  ECOS=systemsmanagement:crowbar:3.0
+              ;;
+              Devel:Cloud:6:SAP)
+                  ECOS=systemsmanagement:crowbar:3.1
+              ;;
+              Devel:Cloud:7)
+                  OSCBUILDDIST=SLE_12_SP2
+                  ECOS=systemsmanagement:crowbar:4.0
+              ;;
+              Devel:Cloud:8)
+                  OSCBUILDDIST=SLE_12_SP3
+                  ECOS=systemsmanagement:crowbar:5.0
+              ;;
+              *) echo "Please add support for the project: $project"
+                  exit 1
           esac
 
+          pkg_filter=`echo ${packagefilter}`
```

Comment:
why this echo? e.g. it will interpret `*` as shell glob which can have unexpected effects.

Reply:
if it contains repeated white spaces, these will be reduced to one - otherwise the next line will produce wrong regexp syntax

Reply:
would still be good to not have bash interpret that - e.g. `pkg_filter=$(echo "$pkg_filter" | sed -e 's/  */ /g')`